pkg/framework/framework.go

```diff
@@ -56,6 +56,23 @@ func GetInfrastructure(c runtimeclient.Client) (*configv1.Infrastructure, error)
 	return infra, nil
 }
 
+var platform configv1.PlatformType
+
+// GetPlatform fetches the PlatformType from the infrastructure object.
+// Caches value after first successful retrieval.
+func GetPlatform(c runtimeclient.Client) (configv1.PlatformType, error) {
+	// platform won't change during test run and might be cached
+	if platform != "" {
+		return platform, nil
+	}
+	infra, err := GetInfrastructure(c)
+	if err != nil {
+		return "", err
+	}
+	platform = infra.Status.PlatformStatus.Type
```

Comment:
PlatformStatus is a pointer, you must guard against a nil pointer before you deref it here

Reply:
I don't feel that i MUST do it there :D 
This thing will crash at the very start of the test suite with a stacktrace which will point right there.

 But good catch, I will add guard, np.